pkg/networkplugin-syncer/handlers/ovn/nbctl/nbctl.go

```diff
@@ -0,0 +1,111 @@
+package nbctl
+
+import (
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strconv"
+
+	"github.com/pkg/errors"
+	"github.com/submariner-io/admiral/pkg/log"
+	"k8s.io/klog"
+
+	"github.com/submariner-io/submariner/pkg/util"
+)
+
+type NbCtl struct {
+	clientKey        string
+	clientCert       string
+	ca               string
+	connectionString string
+}
+
+func New(db, clientkey, clientcert, ca string) *NbCtl {
+	return &NbCtl{
+		connectionString: db,
+		clientKey:        clientkey,
+		clientCert:       clientcert,
+		ca:               ca,
+	}
+}
+
+func (n *NbCtl) nbctl(parameters ...string) (output string, err error) {
+	allParameters := []string{
+		fmt.Sprintf("--db=%s", n.connectionString),
+		"-c", n.clientCert,
+		"-p", n.clientKey,
+		"-C", n.ca,
+	}
+
+	allParameters = append(allParameters, parameters...)
+
+	cmd := exec.Command("/usr/bin/ovn-nbctl", allParameters...)
+
+	klog.V(log.DEBUG).Infof("ovn-nbctl %v", allParameters)
+
+	out, err := cmd.CombinedOutput()
+
+	strOut := string(out)
+
+	if err != nil {
+		klog.Errorf("Error running ovn-nbctl %+v, output:\n%s", err, strOut)
+		return strOut, err
+	}
+
+	return strOut, err
+}
+
+func (n *NbCtl) SetGatewayChassis(lrp, chassis string, prio int) error {
+	_, err := n.nbctl("lrp-set-gateway-chassis", lrp, chassis, strconv.Itoa(prio))
+	return err
+}
+
+func (n *NbCtl) DelGatewayChassis(lrp, chassis string, prio int) error {
+	_, err := n.nbctl("lrp-del-gateway-chassis", lrp, chassis, strconv.Itoa(prio))
+	return err
+}
+
+func (n *NbCtl) GetGatewayChassis(lrp, chassis string) (string, error) {
+	output, err := n.nbctl("lrp-get-gateway-chassis", lrp, chassis)
+	return output, err
+}
+
+func (n *NbCtl) LrPolicyAdd(logicalRouter string, prio int, filter string, actions ...string) error {
+	allParameters := []string{"lr-policy-add", logicalRouter, strconv.Itoa(prio), filter}
+	allParameters = append(allParameters, actions...)
+	_, err := n.nbctl(allParameters...)
+
+	return err
+}
+func (n *NbCtl) LrPolicyDel(logicalRouter string, prio int, filter string) error {
+	_, err := n.nbctl("lr-policy-del", logicalRouter, strconv.Itoa(prio), filter)
+	return err
+}
+
+func (n *NbCtl) LrPolicyGetSubnets(logicalRouter, rerouteIp string) (*util.StringSet, error) {
+	output, err := n.nbctl("lr-policy-list", logicalRouter)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Error getting existing routing policies for router %q", logicalRouter)
```

Comment:
nit: sentence to start with small letters in error messages

Reply:
Done